flaml/ml.py

```diff
@@ -319,8 +319,11 @@ def evaluate_model_CV(
         shuffle = False
     elif isinstance(kf, TimeSeriesSplit):
         kf = kf.split(X_train_split, y_train_split)
-    else:
+    elif hasattr(kf, "split"):
         kf = kf.split(X_train_split)
+    else:

```

Comment:
Use `elif isinstance(kf, list)`. Keep `kf = kf.split(X_train_split)` as default, as there might be other split_type.